src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -128,12 +131,81 @@ public DefinitionRegistrationOptions GetRegistrationOptions(DefinitionCapability
         }
 
-        internal static async Task<TagHelperBinding?> GetOriginTagHelperBindingAsync(
+        private async Task<Range> GetNavigateRangeAsync(DocumentSnapshot documentSnapshot, BoundAttributeDescriptor? attributeDescriptor, CancellationToken cancellationToken)
+        {
+            if (attributeDescriptor is not null)
+            {
+                var originCodeDocument = await documentSnapshot.GetGeneratedOutputAsync().ConfigureAwait(false);
+                var range = await TryGetPropertyRangeAsync(originCodeDocument, attributeDescriptor.GetPropertyName(), _documentMappingService, cancellationToken).ConfigureAwait(false);
+
+                if (range is not null)
+                {
+                    return range;
+                }
+            }
+
+            return new Range(new Position(0, 0), new Position(0, 0));
+        }
+
+        internal static async Task<Range?> TryGetPropertyRangeAsync(RazorCodeDocument codeDocument, string propertyName, RazorDocumentMappingService documentMappingService, CancellationToken cancellationToken)
+        {
+            // Parse the C# file and find the property that matches the name, and that has a [Parameter] attribute
+
+            var csharpText = codeDocument.GetCSharpSourceText();
+            var syntaxTree = CSharpSyntaxTree.ParseText(csharpText, cancellationToken: cancellationToken);
+            var root = await syntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var property = root.DescendantNodes(n => n is BaseNamespaceDeclarationSyntax or CompilationUnitSyntax or ClassDeclarationSyntax)
+                .OfType<PropertyDeclarationSyntax>()
+                .Where(p => p.Identifier.ValueText.Equals(propertyName, StringComparison.Ordinal))
+                .FirstOrDefault();
+
+            // Did we find a property at all?
+            if (property is null)
+            {
+                return null;
+            }
+
+            // We might have found a property, but is it a parameter?
+            foreach (var attributeList in property.AttributeLists)
+            {
+                foreach (var attribute in attributeList.Attributes)
+                {
+                    // Attributes could be simple like [Parameter] or qualified like [Blah.Parameter]
+                    // but we don't care about that distinction, so lets just avoid dealing with the complexities
+                    // of the roslyn tree.
+                    var name = attribute.ToString();
+
+                    // Sadly we don't have symbolic information here, so we have to just hope for the best
+                    // Since we're only navigating its not a big deal, plus its not possible to have multiple
+                    // properties with the same name defined, so its not like one could be a real parameter, and
+                    // another could be a fake one
+                    if (name.Equals("ParameterAttribute", StringComparison.Ordinal) ||
```

Comment:
@TanayParikh are there other cases we should be handling here? `CascadingParameter` comes to mind but I don't think those get translated into HTML attribute equivalents.

Reply:
So stricy speakibg this is all unnecessary anyway, and we could just remove it, rather than chasing a long tail of scenarios.

We know that C# doesn't allow multiple properties with the same name, so if we trust that the component search found the right file, we could just look for a property and ignore attributes. Worst case is someone leaves the attribute off, their razor file has an error, and we go to definition on it anyway... which actually arguably helps them fix the error.

Reply:
Ah that's totally fair too

Reply:
Removed all of this :)